### PR TITLE
Cortex asm pack

### PR DESCRIPTION
**Pull request by andrew-paloalto**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: N/A

## Description
This pack is to add some additional playbook content for working with ASM incident data.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Extract_IP_Indicator.yml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    id: "5"
+    taskid: e7c4ae9f-5b10-4620-8b1d-43d15e2adc6f
+    type: regular
+    task:
+      id: e7c4ae9f-5b10-4620-8b1d-43d15e2adc6f
+      version: -1
+      name: Associate IP Indicator To Alert
+      description: Associate an indicator to a given alert, you can find the indicator
+        to connect either by it's value or it's id
+      script: Builtin|||associateIndicatorToAlert
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Extract_IP_Indicator.yml:166]: @andrew-paloalto  I can't find `associateIndicatorToAlert` automation - is it something you forgot to add or am I missing something?

**Comment by tomer-pan** [comment]
@andrew-paloalto Thank you for the demo, was great!
As discussed I think these are the things we could improve:
1. Change inputs values from this pattern ${value} to filtered values. 
2. Add more vendors to the CMDB + Vulnerabilties playbooks.
3. Add search for vulnerabilities and days number with a playbook input.
5. Same ip on different alerts would run the same playbook several times (v2)
6. Save the ip (and other possible values) to incident fields (alert fields) so it would be available ootb (v2)
7. Add a remediation sub-playbook (v2)

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      type: playbook
+      iscommand: false
+      brand: ""
+      description: ""
+    nexttasks:
+      '#none#':
+      - "2"
+    scriptarguments:
+      AlertName:
+        simple: ${alert.name}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml:55]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      brand: ""
+    nexttasks:
+      "no":
+      - "4"
+      "yes":
+      - "3"
+      - "5"
+    scriptarguments:
+      value:
+        simple: ${ExtractedIndicators.IP}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml:98]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      type: playbook
+      iscommand: false
+      brand: ""
+      description: ""
+    nexttasks:
+      '#none#':
+      - "4"
+    scriptarguments:
+      IP:
+        simple: ${ExtractedIndicators.IP}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml:132]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+      modules:
+        simple: ${ConfiguredModules}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml:134]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+      type: playbook
+      iscommand: false
+      brand: ""
+      description: ""
+    nexttasks:
+      '#none#':
+      - "4"
+    scriptarguments:
+      IP:
+        simple: ${ExtractedIndicators.IP}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml:200]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+      iscommand: false
+      brand: ""
+    nexttasks:
+      "no":
+      - "2"
+      "yes":
+      - "4"
+    scriptarguments:
+      value:
+        simple: ${inputs.IP}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml:141]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+      iscommand: true
+      brand: ServiceNow CMDB
+    nexttasks:
+      '#none#':
+      - "12"
+    scriptarguments:
+      class:
+        simple: cmdb_ci
+      sys_id:
+        simple: ${ServiceNowCMDB.Record.Attributes.cmdb_ci.value}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml:359]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml [hunk 9]
```diff
@@ -0,0 +1,537 @@
...
+      iscommand: false
+      brand: ""
+    nexttasks:
+      "no":
+      - "14"
+      "yes":
+      - "9"
+    scriptarguments:
+      value:
+        simple: ${ServiceNowCMDB.Record.Attributes.cmdb_ci.value}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml:436]: @andrew-paloalto Please change to a filtered value structure.

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml [hunk 10]
```diff
@@ -0,0 +1,537 @@
...
+    nexttasks:
+      '#none#':
+      - "2"
+    scriptarguments:
+      columns:
+        simple: name,sys_class_name,sys_id,asset.display_value,asset.link,asset.value
+      data:
+        simple: |2-
+
+          ${ServiceNowCMDB.Record.Attributes}
```
Comment (tomer-pan) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_CMDB_Enrichment.yml:397]: @andrew-paloalto Please change to a filtered value structure (and remove the unneeded new line prefix).

**Comment by yucohen** [comment]
@andrew-paloalto Hi. this error is from the validate step:
> [ERROR]: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml: [PB117] - Playbook Cortex ASM - ASM Alert uses ['Cortex ASM - ServiceNow CMDB Enrichment'], which do not exist in the id_set.

did you mean `Cortex ASM - CMDB Enrichment`? please take a look at this

**Comment by andrew-paloalto** [comment]
> @andrew-paloalto Hi. this error is from the validate step:
> 
> > [ERROR]: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml: [PB117] - Playbook Cortex ASM - ASM Alert uses ['Cortex ASM - ServiceNow CMDB Enrichment'], which do not exist in the id_set.
> 
> did you mean `Cortex ASM - CMDB Enrichment`? please take a look at this

Fixed, thanks!

**Review by yucohen — APPROVED** [review]
handle the validations in the internal

**Pull request merged**